shm/Cargo.hpp

```diff
@@ -4,22 +4,21 @@
 
 class Cargo {
 public:
-    Cargo(const std::string& name,
-          size_t amount,
-          size_t basePrice)
-        : _name(name), _amount(amount), _basePrice(basePrice) {}
+    Cargo(const std::string& name, size_t amount, size_t basePrice);
+    virtual ~Cargo() = default;
 
-    Cargo& operator+=(const size_t amount);
-    Cargo& operator-=(const size_t amount);
+    virtual size_t getPrice() const = 0;
+    virtual std::string getName() const = 0;
+    virtual size_t getAmount() const = 0;
+    virtual size_t getBasePrice() const = 0;
+    //virtual size_t getMaxAmount() const = 0;
 
-    bool operator==(const Cargo& src) const;
-    bool operator!=(const Cargo& src) const;
+    virtual Cargo& operator+=(const size_t amount) = 0;
+    virtual Cargo& operator-=(const size_t amount) = 0;

```

Comment:
I think these operators definition will be the same for each cargo item. How can we solve it?  Maybe Cargo should be abstract class instead of interface.

Reply:
+1

Reply:
I agree with that. These methods are the same implementation in derived classes.